packages/metadata/test/unit/providers/githubMetadata.provider.spec.ts

```diff
@@ -0,0 +1,176 @@
+import MockAdapter from "axios-mock-adapter";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+import { ILogger, ZKChainMetadataItem } from "@zkchainhub/shared";
+
+import { FetchError, InvalidSchema } from "../../../src/internal";
+import { GithubMetadataProvider } from "../../../src/providers/githubMetadata.provider";
+import {
+    chainJsonUrl,
+    mockChainData,
+    mockTokenData,
+    tokenJsonUrl,
+} from "../../fixtures/metadata.fixtures";
+
+const mockLogger: ILogger = {
+    info: vi.fn(),
+    warn: vi.fn(),
+    error: vi.fn(),
+    debug: vi.fn(),
+};
+
+describe("GithubMetadataProvider", () => {
+    beforeEach(() => {
+        vi.resetAllMocks();
+    });
+
+    describe("getChainsMetadata", () => {
+        it("should return the chains metadata", async () => {
+            const provider = new GithubMetadataProvider(tokenJsonUrl, chainJsonUrl, mockLogger);
+            const axios = provider["axios"];
+            const axiosGetMock = vi
+                .spyOn(axios, "get")
+                .mockResolvedValueOnce({ data: mockChainData });
+
+            const result = await provider.getChainsMetadata();
+
+            expect(axiosGetMock).toHaveBeenCalledWith(chainJsonUrl);
+            expect(result).toBeInstanceOf(Map);
+            expect(result.size).toBe(2);
+
+            const chain1 = result.get(324n) as ZKChainMetadataItem;
+            expect(chain1).toBeDefined();
+            expect(chain1.chainId).toBe(324n);
+            expect(chain1.name).toBe("ZKsyncERA");
+            expect(chain1.iconUrl).toBe(
+                "https://s2.coinmarketcap.com/static/img/coins/64x64/24091.png",
+            );
+            expect(chain1.publicRpcs).toEqual([
+                "https://mainnet.era.zksync.io",
+                "https://zksync.drpc.org",
+                "https://zksync.meowrpc.com",
+            ]);
+            expect(chain1.explorerUrl).toBe("https://explorer.zksync.io/");
+            expect(chain1.launchDate).toBe(1679626800);
+            expect(chain1.chainType).toBe("Rollup");
+            expect(chain1.baseToken).toEqual({
+                name: "Ethereum",
+                symbol: "ETH",
+                coingeckoId: "ethereum",
+                type: "native",
+                contractAddress: null,
+                decimals: 18,
+                imageUrl:
+                    "https://coin-images.coingecko.com/coins/images/279/large/ethereum.png?1696501628",
+            });
```

Comment:
What's your take on using a `toMatchObject` matcher instead of one expect per property? Yay or nay?

Reply:
cc @0xkenj1

Reply:
when you tab 'copilot suggestions' xd, makes sense using `toMatchObject`